Application/EdFi.Ods.Common/Providers/Queries/AggregateRootCriteriaProviderHelpers.cs

```diff
@@ -26,4 +30,36 @@ public static string[] GetUniqueIdProperties(IPersonTypesProvider personTypesPro
         "SortBy",
         "SortDirection"
     };
+    
+    public static EntityProperty FindIdentificationCodeProperty(Entity entity)
+    {
+        var entityFullName = entity.FullName.Name;
+        var identificationCodeEntityProperty = _entityIdentificationCodePropertyCache.GetValueOrDefault(entityFullName);
+
+        if (identificationCodeEntityProperty != null)
+            return identificationCodeEntityProperty;
+
+        while (entity != null)
+        {
+            identificationCodeEntityProperty = FindIdentificationCodePropertyInAssociations(entity);
+
+            if (identificationCodeEntityProperty != null)
+            {
+                _entityIdentificationCodePropertyCache[entityFullName] = identificationCodeEntityProperty;
+                return identificationCodeEntityProperty;
+            }
+
+            entity = entity.BaseEntity;
+        }
+
+        return null;
+
+        EntityProperty FindIdentificationCodePropertyInAssociations(Entity entity)
```

Comment:
This method name is misleading because you're looking for a property in the child table, not the association (aka FK). The identification code will never be part of any association. So if we were leaving this as-is (but I recommend changes to what this is returning below), I would say call this "TryFindIdentificationCodeProperty" and change the semantics (return a `bool` for success/failure, and an `out` property). But... read on.